tests/integration/test_s3.py

```diff
@@ -198,6 +198,19 @@ def test_invalid_range_error(self):
         # clean up
         self._delete_bucket(bucket_name, ['steve'])
 
+    def test_range_key_not_exists(self):
+        bucket_name = 'range-%s' % short_uid()
+        self.s3_client.create_bucket(Bucket=bucket_name)
+
+        self.s3_client.create_bucket(Bucket=bucket_name)
+        try:
+            self.s3_client.get_object(Bucket=bucket_name, Key='key', Range='bytes=1024-4096')
```

Comment:
nit: Let's better use `self.assertRaises` here - see here for an example: https://github.com/localstack/localstack/blob/bf672b0cd67acd1bfb56967ca3077ff8d2b26d78/tests/integration/test_dynamodb.py#L395-L397